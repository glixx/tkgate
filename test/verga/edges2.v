//
// Non-blocking assignment test.
//
module top;
  reg x;
  supply1 vcc;
  supply0 gnd;

  always @(posedge x) $display("posedge x=%d",x);
  always @(negedge x) $display("negedge x=%d",x);

  initial
    begin
	$display("x=%d vcc=%d gnd=%d",x,vcc,gnd);
	vcc = 0;
	$display("x=%d vcc=%d gnd=%d",x,vcc,gnd);
	#1 x = vcc;
	#1 x = gnd;
	#1 x = vcc;
	#1 x = gnd;
	#1 x = gnd;
	#1 $display("done");
    end

endmodule

