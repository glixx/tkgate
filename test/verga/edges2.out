x=x vcc=1 gnd=0
x=x vcc=1 gnd=0
posedge x=1
negedge x=0
posedge x=1
negedge x=0
done
